### PR TITLE
similarity calculation coefficients are now dynamic

### PR DESCRIPTION
**Pull request by bestDvlpr**
similarity calculation coefficients are now taken from status request to compreface-core part

### CODE REVIEW

File: java/api/src/main/java/com/exadel/frs/core/trainservice/component/classifiers/EuclideanDistanceClassifier.java [hunk 1]
```diff
@@ -109,7 +108,13 @@ private INDArray normalize(final INDArray embeddings) {
     }
 
     private INDArray calculateSimilarities(INDArray distance) {
-        return tanh(distance.rsubi(1.1).muli(2.5), false).addi(1).divi(2);
+        FacesStatusResponse status = facesApiClient.getStatus();
```
[the remark this answers is not in the document]
Comment (bestDvlpr) [reply java/api/src/main/java/com/exadel/frs/core/trainservice/component/classifiers/EuclideanDistanceClassifier.java:111]: Inside `getStatus` implementation, if something wrong, it is throwing an exception.

File: java/api/src/main/java/com/exadel/frs/core/trainservice/service/FaceService.java [hunk 2]
```diff
@@ -82,6 +82,7 @@ public FaceBO findAndSaveFace(
             final Double detProbThreshold,
             final String modelKey
     ) throws IOException {
+        facesApiClient.getStatus();
```
[the remark this answers is not in the document]
Comment (bestDvlpr) [reply java/api/src/main/java/com/exadel/frs/core/trainservice/service/FaceService.java]: I have removed unneeded `getStatus()` invocations.

File: java/api/src/main/java/com/exadel/frs/core/trainservice/service/FaceRecognizeProcessServiceImpl.java [hunk 3]
```diff
@@ -21,12 +24,14 @@
...
 @Service("recognitionService")
 @RequiredArgsConstructor
+@Slf4j
 public class FaceRecognizeProcessServiceImpl implements FaceProcessService {
 
     private final FaceClassifierPredictor classifierPredictor;
     private final FacesApiClient client;
     private final ImageExtensionValidator imageValidator;
     private final FacesMapper mapper;
+    private final CacheManager cacheManager;
```
Comment (pospielov) [inline java/api/src/main/java/com/exadel/frs/core/trainservice/service/FaceRecognizeProcessServiceImpl.java:34]: Do we need it here?

File: java/api/src/main/java/com/exadel/frs/core/trainservice/sdk/faces/service/FacesRestApiClient.java [hunk 4]
```diff
@@ -24,6 +25,7 @@
     @Override
     public FindFacesResponse findFaces(final MultipartFile photo, final Integer faceLimit, final Double thresholdC,
                                        final String facePlugins) {
+        getStatus();
```
Comment (pospielov) [inline java/api/src/main/java/com/exadel/frs/core/trainservice/sdk/faces/service/FacesRestApiClient.java:28]: do we need it here?

File: java/api/src/main/java/com/exadel/frs/core/trainservice/component/migration/MigrationComponent.java [hunk 5]
```diff
@@ -63,7 +67,7 @@ public void migrate(final String url) {
     private void processFaces(final String url) {
         val migrationServerFeignClient = feignClientFactory.getFeignClient(FacesFeignClient.class, url);
         val migrationCalculatorVersion = migrationServerFeignClient.getStatus().getCalculatorVersion();
-
+        facesRestApiClient.getStatus();
```
Comment (pospielov) [inline java/api/src/main/java/com/exadel/frs/core/trainservice/component/migration/MigrationComponent.java:70]: do we need it here?

File: java/api/src/main/java/com/exadel/frs/core/trainservice/component/classifiers/EuclideanDistanceClassifier.java [hunk 6]
```diff
@@ -113,7 +113,13 @@ private INDArray normalize(final INDArray embeddings) {
...
+        if (status == null || status.getSimilarityCoefficients() == null || status.getSimilarityCoefficients().isEmpty()) {
+            throw new FacesServiceException("No status information received");
```
Comment (pospielov) [inline java/api/src/main/java/com/exadel/frs/core/trainservice/component/classifiers/EuclideanDistanceClassifier.java:118]: Could you change it to:
Error during synchronization between servers: No status information received

Comment (bestDvlpr) [reply java/api/src/main/java/com/exadel/frs/core/trainservice/component/classifiers/EuclideanDistanceClassifier.java:118]: Indeed the exception constructor creates error response with the message _Error during synchronization between servers: No status information received_

**Review by pospielov — APPROVED** [review]

**Pull request merged**